### ISSUE TITLE
Add `AllowTrailingComment` configuration option to `Style/DisableCopsWithinSourceCodeDirective`

### ISSUE THREAD

**Issue by sanfrecce-osaka** [rubocop/rubocop#15073]
### Before submitting

- [x] I have searched the existing issues and this hasn't been requested yet.

### Cop name

Style/DisableCopsWithinSourceCodeDirective

### Is your feature request related to a problem?

Once `rubocop:disable` comments start appearing in a codebase, they tend to proliferate without bound, undermining the value of static analysis. With the increasing adoption of AI agents for code generation, this tendency can accelerate even further.

`Style/DisableCopsWithinSourceCodeDirective` can help address this problem, but the existing `AllowedCops` option only permits disabling specific cops. Whether a `rubocop:disable` directive should be tolerated often depends on context rather than the cop being disabled, and `AllowedCops` alone cannot cover these cases.

RuboCop officially supports a trailing comment syntax using `--` on directive comments. An option that treats the presence of such a trailing comment as an indication that the suppression context has been documented would allow teams to express more flexible policies through configuration.

### Describe the solution you'd like

Add an `AllowTrailingComment` option to `Style/DisableCopsWithinSourceCodeDirective`.

When set to `true`, directives that include a trailing comment using the `--`
marker are not flagged as offenses.

### Code examples

```ruby
# bad
# rubocop:disable Metrics/AbcSize
def foo
end
# rubocop:enable Metrics/AbcSize

# bad
# rubocop:disable Metrics/AbcSize --
def foo
end
# rubocop:enable Metrics/AbcSize

# good
# rubocop:disable Metrics/AbcSize -- very complex domain
def foo
end
# rubocop:enable Metrics/AbcSize
```

### Additional context

- Other issues related to `Style/DisableCopsWithinSourceCodeDirective`
  - https://github.com/rubocop/rubocop/issues/15042
- Similar discussion
  - https://github.com/rubocop/rubocop/discussions/12552
  - This issue applies only to trailing comments that follow directive comments.
